server/interpreter.js

```diff
@@ -2085,7 +2097,7 @@ Interpreter.prototype['stepAssignmentExpression'] = function() {
   var node = state.node;
   if (!state.doneLeft_) {
     state.doneLeft_ = true;
-    stack.push({node: node['left'], components: true});
+    this.pushNode_(node['left']).components = true;
```

Comment:
The goal of this PR is to speed up the interpreter.  Adding a for-in loop inside the interpreter's second-most called function adds about 150ms to the test.  The PR is still a net improvement, but it's a pretty steep cost.

Alternatively, we could spilt the lines in two:
`var node = this.pushNode_(node['foo']);`
`node.throwValue = value;`